cmd/repo-updater/repos/github.go

```diff
@@ -174,180 +174,275 @@ func (s *GithubSource) excludes(r *github.Repository) bool {
-					break
-				}
-
-				hasNextPage = reposPage.HasNextPage
-				repos := reposPage.Repos
-
-				rateLimitRemaining, rateLimitReset, rateLimitRetry, _ := s.searchClient.RateLimit.Get()
-				log15.Debug(
-					"github sync: ListRepositoriesForSearch",
-					"searchString", repositoryQuery,
-					"repos", len(repos),
-					"rateLimitRemaining", rateLimitRemaining,
-					"rateLimitReset", rateLimitReset,
-					"retryAfter", rateLimitRetry,
-				)
-
-				for _, r := range repos {
-					set[r.DatabaseID] = r
-				}
-
-				if hasNextPage {
-					// GitHub search has vastly different rate limits to
-					// the normal GitHub API (30req/m vs
-					// 5000req/h). RecommendedWaitForBackgroundOp has
-					// heuristics tuned for the normal API, part of which
-					// is to not sleep if we have ample rate limit left.
-					//
-					// So we only let the heuristic kick in if we have
-					// less than 5 requests left.
-					remaining, _, retryAfter, ok := s.searchClient.RateLimit.Get()
-					if retryAfter > 0 || (ok && remaining < 5) {
-						time.Sleep(s.searchClient.RateLimit.RecommendedWaitForBackgroundOp(1))
-					}
-				}
-			}
+		pageRepos, hasNextPage, wait, err := pager(page)
+		if err != nil {
+			break
+		}
+
+		for _, r := range pageRepos {
+			set[r.DatabaseID] = r
+		}
+
+		if hasNextPage && wait > 0 {
+			time.Sleep(s.client.RateLimit.RecommendedWaitForBackgroundOp(wait))
 		}
 	}
 
-	err := s.fetchAllRepositoriesInBatches(ctx, set)
-	if err != nil {
+	return set, nil
+}
+
+// listOrg handles the `org` config option.
+// It returns all the repositories belonging to the given organization
+// by hitting the /orgs/:org/repos endpoint.
+func (s *GithubSource) listOrg(ctx context.Context, org string) (map[int64]*github.Repository, error) {
+	return s.paginate(
+		ctx,
+		func(page int) (repos []*github.Repository, hasNextPage bool, rateLimitCost int, err error) {
+			repos, hasNextPage, rateLimitCost, err = s.client.ListOrgRepositories(ctx, org, page)
+			rateLimitRemaining, rateLimitReset, rateLimitRetry, _ := s.client.RateLimit.Get()
+			log15.Debug(
+				"github sync: ListOrgRepositories",
+				"repos", len(repos),
+				"rateLimitCost", rateLimitCost,
+				"rateLimitRemaining", rateLimitRemaining,
+				"rateLimitReset", rateLimitReset,
+				"retryAfter", rateLimitRetry,
+			)
+			return
+		},
+	)
+}
+
+// listRepos returns the valid repositories from the given list of repository names.
+// This is done by hitting the /repos/:owner/:name endpoint for each of the given
+// repository names.
+func (s *GithubSource) listRepos(ctx context.Context) (map[int64]*github.Repository, error) {
```

Comment:
Similarly to what we do in the other `listXXX` methods, we should probably parameterise the list of repos to be listed here.

```suggestion
func (s *GithubSource) listRepos(ctx context.Context, repos ...string) (map[int64]*github.Repository, error) {
```